src/coreclr/src/nativeaot/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.Windows.cs

```diff
@@ -126,23 +126,15 @@ private bool SetPriorityLive(ThreadPriority priority)
             return Interop.Kernel32.SetThreadPriority(_osHandle, (int)MapToOSPriority(priority));
         }
 
-        private ThreadState GetThreadState()
+        [UnmanagedCallersOnly]
+        private static void OnThreadExit()
         {
-            int state = _threadState;
-            // If the thread is marked as alive, check if it has finished execution
-            if ((state & (int)(ThreadState.Unstarted | ThreadState.Stopped | ThreadState.Aborted)) == 0)
+            Thread currentThread = t_currentThread;
+            if (currentThread != null)
             {
-                if (JoinInternal(0))
-                {
-                    state = _threadState;
-                    if ((state & (int)(ThreadState.Stopped | ThreadState.Aborted)) == 0)
-                    {
-                        SetThreadStateBit(ThreadState.Stopped);
-                        state = _threadState;
-                    }
-                }
+                StopThread(currentThread);
+                currentThread.JoinInternal(0);
```

Comment:
Why is the JoinInternal needed here?

Reply:
TLDR: I did it at 2AM 😂 
Probably I just followed the logic we have in Unix. I see that in Unix version wait object is managed and we need to signal it once we got that callback, at the same time on Windows signaling is done with OS handle by kernel mechanisms related to those that invoked our `OnThreadExit`. So yes, completely not needed.